code/modules/lighting/lighting_system.dm

```diff
@@ -246,10 +246,27 @@
 	anchored = 1
 
 /atom/movable/light/Destroy(force)
-	if(force)
-		. = ..()
-	else
+	if(!force)
 		return QDEL_HINT_LETMELIVE
+	var/turf/T = loc
+	. = ..()
+	if (T.lighting_object == src)
+		T.lighting_object = null
+	
+/atom/movable/light/New()
+	if (!loc || !isturf(loc))
```

Comment:
I was in the middle of typing out a comment about how my way was faster, but then I realized I was having a total brain fart.

Technically !loc is faster when it's null, (by like 4 cpu ticks, like it matters) but its much slower when it's not null since it still has to !isturf(), and my brain hadn't pieced that second part up for some reason